test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs

```diff
@@ -1900,5 +1901,41 @@ private static string GetTemporaryDirectory()
             Directory.CreateDirectory(tempDirectory);
             return tempDirectory;
         }
+
+        [TensorFlowFact]
+        public void TensorflowPlaceholderShapeInferenceTest()
+        {
+            //download the pb file
+            string gitPath = "https://github.com/dotnet/machinelearning-testdata/raw/ddf00f8d60ff1b4bb1f3e33639bec318e67fd32e/Microsoft.ML.TensorFlow.TestModels/cifar_model/frozen_model_variadic_input_shape.pb";
+            using (WebClient client = new WebClient())
+            {
```

Comment:
It looks like the other tests are using Microsoft.ML.TensorFlow.TestModels to store and download their models. That may be the standard way to do it, but it would require us to update that package to include the new model you care about.

That model is published here: [https://dotnet.myget.org/feed/dotnet-core/package/nuget/Microsoft.ML.TensorFlow.TestModels](https://dotnet.myget.org/feed/dotnet-core/package/nuget/Microsoft.ML.TensorFlow.TestModels) but its source doesn't seem to be in our source code.

@codemzs / @eerhardt: Can you please help on how this package should be updated if we wish to include a new model?

Reply:
You can create the package from the `dotnet/machinelearning-testdata` repo by [using nuget.exe to pack ](https://docs.microsoft.com/en-us/nuget/create-packages/creating-a-package#run-nuget-pack-to-generate-the-nupkg-file) the Microsoft.ML.TensorFlow.TestModels.nuspec file. Note you also need to bump the version of the package [here](https://github.com/dotnet/machinelearning-testdata/blob/master/Microsoft.ML.TensorFlow.TestModels.nuspec#L5).

Then once you have the `.nupkg` created, you can upload it to a NuGet feed. Note that we are moving away from myget.org, and the new testdata feed is:

https://github.com/dotnet/machinelearning/blob/fa54370294303e0ccd0a0c30e3aa645689b7db69/Directory.Build.props#L27

Reply:
Thanks @harishsk and @eerhardt! I have generated the nupkg but have trouble uploading it. If I upload it using [Web portal](https://www.nuget.org/packages/manage/upload). I get "This package ID has been reserved. Please request access to upload to this reserved namespace from the owner of the reserved prefix, or re-upload the package with a different ID". I also tried nuget push but being asked "Please provide credentials for: https://pkgs.dev.azure.com/dnceng/public/_packaging/machinelearning-testdata/nuget/v3/index.json". Am I missing something?

Reply:
It is possible you do not have permissions. I have built and uploaded the package. 

Thanks @eerhardt